morphir/src/org/finos/morphir/runtime/sdk/SetSDK.scala

```diff
@@ -1,23 +1,48 @@
 package org.finos.morphir.runtime.sdk
 
 import org.finos.morphir.ir.Type
-import org.finos.morphir.runtime._
-import org.finos.morphir.runtime.internal._
+import org.finos.morphir.runtime.*
+import org.finos.morphir.runtime.internal.*
 import org.finos.morphir.runtime.RTValue
+import org.finos.morphir.runtime.RTValue.coerceComparable
 
 import scala.collection.mutable
 
 object SetSDK {
 
+  val empty: SDKValue = SDKValue.SDKNativeValue(RTValue.Set(mutable.LinkedHashSet.empty))
+
   val foldr = DynamicNativeFunction3("foldr") {
     (context: NativeContext) => (f: RTValue.Function, zero: RTValue, set: RTValue.Set) =>
-      // TODO: change foldr to sort before folding.  Currently blocked on comparable support
+      val sortedSet = sortSet(set)
       // Elm foldr iterates on sort order, Scala iterates on insertion order.  Sort before folding.
-      set.value.foldRight(zero) { (a, acc) =>
+      sortedSet.foldRight(zero) { (a, acc) =>
         context.evaluator.handleApplyResult2(Type.UType.Unit(()), f, a, acc)
       }
   }
 
+  val foldl = DynamicNativeFunction3("foldl") {
+    (context: NativeContext) => (f: RTValue.Function, zero: RTValue, set: RTValue.Set) =>
+      val sortedSet = sortSet(set)
+      // Elm foldl iterates on sort order, Scala iterates on insertion order.  Sort before folding.
+      sortedSet.foldLeft(zero) { (acc, a) =>
```

Comment:
Given how much trouble the order of arguments was, I'd leave a comment explaining it

Reply:
will add on my next pr 👍